src/master-playlist-controller.js

```diff
@@ -793,6 +802,46 @@ export class MasterPlaylistController extends videojs.EventTarget {
   }
 
   /**
+   * Check if a playlist has stopped being updated
+   * @param {Object} playlist the media playlist object
+   * @return {boolean} whether the playlist has stopped being updated or not
+   */
+  isPlaylistOutdated_(playlist) {
```

Comment:
I know I've suggested renaming this already, but now that we have an event that signifies the playlist is outdated/not refreshing, and this method more checks if we are stuck at the end of the playlist, so maybe somthing like `stuckAtPlaylistEnd_`or `isAtPlaylistEnd_`. Then our logic sort of semantically reads like if the refreshed playlist is unchanged and the player is at the end of the playlist, blacklist this playlist.